app/controllers/concerns/signature_verification.rb

```diff
@@ -180,11 +189,18 @@ def verify_signature(actor, signature, compare_signed_string)
     nil
   end
 
-  def build_signed_string
+  def build_signed_string(include_query_string: true)
     signed_headers.map do |signed_header|
       case signed_header
       when Request::REQUEST_TARGET
-        "#{Request::REQUEST_TARGET}: #{request.method.downcase} #{request.path}"
+        if include_query_string
+          "#{Request::REQUEST_TARGET}: #{request.method.downcase} #{request.original_fullpath}"
+        else
+          # Older versions of Mastodon incorrectly omit the query string from the (request-target) pseudo-header.
```

Comment:
We could probably add the version numbers that had this behaviour here?

Reply:
Well, those are all Mastodon versions for now.